src/pfe/file-watcher/server/src/projects/ShellExtensionProject.ts

```diff
@@ -208,26 +234,35 @@ export class ShellExtensionProject implements IExtensionProject {
 
     /**
      * @function
-     * @description Get the build log for the project.
+     * @description Get logs from files or directories.
      *
-     * @param logDirectory <Required | string> - The log location directory.
+     * @param type <Required | String> - The type of log ("build" or "app")
+     * @param logDirectory <Required | String> - The log location directory.
+     * @param projectID <Required | String> - An alphanumeric identifier for a project.
+     * @param containerName <Required | String> - The docker container name.
      *
-     * @returns Promise<BuildLog>
+     * @returns Promise<Array<AppLog | BuildLog>>
      */
-    getBuildLog = async (logDirectory: string): Promise<BuildLog> => {
-        return await logHelper.getBuildLogs(logDirectory, this.config.buildLogs);
+    getLogs = async (type: string, logDirectory: string, projectID: string, containerName: string): Promise<Array<AppLog | BuildLog>> => {
+        if (type.toLowerCase() != "build" && type.toLowerCase() != "app") return;
+        this.setLogsOriginFromExtension();
```

Comment:
I think we discussed that this could be set right after we load the config since it won't change, so we don't need to run this every time `getLogs` is called.

Reply:
Sure, that makes sense.